api/app/controllers/mno_enterprise/jpi/v1/admin/users_controller.rb

```diff
@@ -182,7 +203,30 @@ def update_sub_tenant(user)
     end
 
     def authorize_user_for_support_role
-      authorize! :read, @user if current_user.support?
+      (authorize! :read, @user) if current_user.support?
+    end
+
+    def valid_support_search
+      return head :forbidden if support_user_index_params.empty?
+    end
+
+    def user_index_params
+      current_user.support? ? support_user_index_params : params
+    end
+
+    def support_user_index_params
+      # Only allow support users to search by first name and last name.
+      params.permit(:terms).tap do |whitelisted|
+        term_params = {}
+
+        first_name_like = params[:terms].present? && JSON.parse(params[:terms])["name.like"]
```

Comment:
You should extract this `params[:terms].present? && JSON.parse(params[:terms])` to avoid repetition